### DISCUSSION TITLE
Using pymeshlab with task scheduler gives GLEW error

### DISCUSSION THREAD

**Discussion by Totil-a** [cnr-isti-vclab/meshlab#933] [Q&A]
Hi there,

I want to use pymeshlab (love it so far) to do some tasks over night.

But when I start a task with windows task scheduler, it says:

pymeshlabpymeshlab.pmeshlab.PyMeshLabException: Failed to apply filter: ambient_occlusion
Details: GLEW initialization failed: Missing GL version

Running my python script 'onscreen', i.e. manually, it all works fine so I assume it has something to do with it being started off screen via task scheduler. But totally unsure how to solve that (if that is even possible).

Many thanks for a response in advance.

Kind regards
Karsten

**Comment by alemuntoni** [comment]
Do you mean windows task scheduler? 
I never used it, and I really don't know how it works and if it allows to run generic OpenGL tasks.
Ambient occlusion is one of the filters that require a working OpenGL context to be executed, and apparently it cannot even be initialized.

**Reply by alemuntoni** [reply]
You could try to execute the following line after importing pymeshlab:

```
pymeshlab.use_cpu_opengl()
```
It will rename a opengl dll inside the pymeshlab package that will be used instead of the system one, that maybe is missing when executing using the scheduler. Hardware acceleration will be disabled when using pymeshlab.

**Reply by Totil-a** [reply]
Thanks for your reply.

"Do you mean windows task scheduler?"
Yes I meant windows task scheduler. A shame of the OpenGL context is not available. That would have been great. You think there is any way to have the OpenGL context available even when using windows task scheduler?

When you say "Hardware acceleration will be disabled when using pymeshlab." does that when called pymeshlab.use_cpu_opengl() before or always?

**Reply by alemuntoni** [reply]
Right now it is always, because the `use_cpu_opengl()` just renames a file, and that file will stay renamed. There is also the `do_not_use_cpu_opengl()` to undo the operation.

Unfortunately this is how it works this opengl stuff on windows...

**Reply by Totil-a** [reply]
I tried it with use_cpu_opengl() as well but it all only works if I am logged on to Windows meaning I have to specify within the windows task scheduler that the task is only to be executed when the user is logged on. In that case windows creates a command window and all works as expected when executing the task.

If, on the other hand, the task is set to 'Run whether user is logged on or not' the script fails. (no shell is being created, not sure if those things are related).
